Sources/AST/ASTVisitor/ASTVisitor.swift

```diff
@@ -521,10 +520,12 @@ public struct ASTVisitor {
     var passContext = passContext
     var processResult = pass.process(doCatchStatement: doCatchStatement, passContext: passContext)
 
+    processResult.passContext.isInsideDoBlock = true
```

Comment:
Nesting? How would this work?
Why not have a do block context with an inner context and add one context to another (outer context) and unwrap when you unwrap the stack?